pkg/providers/openstack/stage.go

```diff
@@ -54,20 +54,22 @@ func (p *OpenstackProvider) Stage(params types.StageImageParams) (_ *types.Image
 		return nil, errors.New("statting raw image file", err)
 	}
 
-	// TODO: Obtain image LOGICAL size, not actual (e.g. 10GB for OSv, not 8MB)
-	imageSizeB := rawImageFile.Size()
-	imageSizeMB := int(unikos.Bytes(imageSizeB).ToMegaBytes())
+	// Image size - use logical size unless physical size is bigger
+	imagePhysicalSizeMB := int(unikos.Bytes(rawImageFile.Size()).ToMegaBytes())
+	if imagePhysicalSizeMB > params.RawImage.RunSpec.MinInstanceDiskMB {
+		params.RawImage.RunSpec.MinInstanceDiskMB = int(imagePhysicalSizeMB)
+	}
 
 	// Pick flavor.
-	flavor, err := pickFlavor(clientNova, imageSizeMB, 0)
+	flavor, err := pickFlavor(clientNova, params.RawImage.RunSpec.MinInstanceDiskMB, 0)
```

Comment:
Yes, it would make sense. But it cannot be done easily. UniK separates compilers and providers very strictly so at the time we are building unikernel (with Capstan), UniK is not aware of available flavors yet.

I'd leave this for now. With manifest.yaml we allow user to set arbitrary logical size and at this point I'd leave it to user to pick the size that matches flavor.